pandas/io/formats/style.py

```diff
@@ -541,6 +723,32 @@ def _translate(self):
 
         return d
 
+    def _translate_latex(self, d: Dict) -> Dict:
+        # post processing of d for latex template:
+        # - remove hidden columns from the non-index part of the table
+        # - place cellstyles in individual td cells
+        # - reinsert missing index th in row if part of multiindex for multirow
+        d["head"] = [[col for col in row if col["is_visible"]] for row in d["head"]]
+        body = []
+        for r, row in enumerate(d["body"]):
+            row_body_cells = [  # replicate the td cells, give them cellstyle
+                {**col, "cellstyle": self.ctx[r, c - self.data.index.nlevels]}
+                for c, col in enumerate(row)
+                if (col["is_visible"] and col["type"] == "td")
+            ]
+            row_body_headers = [  # if a th element is not visible due to multiindex
+                {  # then reinsert it for latex structuring
+                    **col,
```

Comment:
A `th` element might also be not visible due to hidden columns. This doesn't differentiate between the two - needs a fix I think.

Reply:
a new test demonstrates there was no fault here, in fact.